### ISSUE TITLE
Namespacing support (PR proposal)

### ISSUE THREAD

**Issue by abruere** [learningequality/vue-intl#14]
I’m working on namespaced message keys `namespace.translationKey` from a messages json like this one: 
```js
{
  namespace: {
    translationKey: "translated"
  }
}
```
Currently, namespacing can be faked by flattening the json before loading with `Vue.registerMessages`.

As this has been discussed in other formatJS ports where default behavior is deep namespacing ([react](https://github.com/yahoo/react-intl/issues/46), [ember](https://github.com/ember-intl/ember-intl/issues/24)), it should not be the only way to support namespacing, but an alternative one.

Though, current users of this repo could by relying on dot namespacing in keys. We could support this as fallback if `namespace` is not an object as in the example above, but `namespace.translationKey` is defined: 
```js
{
  "namespace.translationKey": "translated"
}
```
Thanks for this formatJS port anyway! I’ll be glad to submit a PR.

**Comment by rtibbles** [comment]
Thank you for the issue, and the PR, will review shortly!

**Comment by rtibbles** [comment]
This has now been released in v3.1.0 - thank you again for your contribution!

**Comment by abruere** [comment]
Awesome, thanks!
